record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/RecordBuilderMetaDataLoader.java

```diff
@@ -17,23 +17,34 @@
 
 import io.soabase.recordbuilder.core.RecordBuilderMetaData;
 
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
 import java.util.function.Consumer;
 
+import javax.annotation.processing.ProcessingEnvironment;
+
 class RecordBuilderMetaDataLoader {
     private final RecordBuilderMetaData metaData;
 
-    RecordBuilderMetaDataLoader(String metaDataClassName, Consumer<String> logger) {
-        RecordBuilderMetaData localMetaData = null;
+    RecordBuilderMetaDataLoader(ProcessingEnvironment processingEnv, Consumer<String> logger) {
+        Map<String, String> options = processingEnv.getOptions();
+        String metaDataClassName = options.get(RecordBuilderMetaData.JAVAC_OPTION_NAME);
         if ((metaDataClassName != null) && !metaDataClassName.isEmpty()) {
+            RecordBuilderMetaData loadedMetaData = null;
             try {
                 Class<?> clazz = Class.forName(metaDataClassName);
-                localMetaData = (RecordBuilderMetaData) clazz.getDeclaredConstructor().newInstance();
-                logger.accept("Found meta data: " + localMetaData.getClass());
+                loadedMetaData = (RecordBuilderMetaData) clazz.getDeclaredConstructor().newInstance();
+                logger.accept("Found meta data: " + clazz);
+            } catch (InvocationTargetException e) {
+                // log the thrown exception instead of the invocation target exception
+                logger.accept("Could not load meta data: " + metaDataClassName + " - " + e.getCause());
             } catch (Exception e) {
-                logger.accept("Could not load meta data: " + metaDataClassName + " - " + e.getMessage());
+                logger.accept("Could not load meta data: " + metaDataClassName + " - " + e);
```

Comment:
I've removed `.getMessage()` call from the exception, as printing the exception information with the type included can be more informational for the user.